### ISSUE TITLE
False surrounding vehicles scan order for SSM device

### ISSUE THREAD

**Issue by leoluecken** [eclipse/sumo#5644]
Scanning recursively for all upstream vehicles may confuse some downstream conflicts since the ego conflict lane is not recognized correctly (but set to its current lane). Downstream scan along the ego's route should complete before executing upstream scans.

**Comment by m-kro** [comment]
In turn, this solution may as well cause wrong ego conflict lanes itself due to the scan order, see #5527

**Comment by m-kro** [comment]
Currently, all internal lanes of a junction get assigned the same ego conflict lane, even if they are not foes of the ego route. If such a lane leads to a separate crossing with the ego route, still the internal lane of the other intersection is chosen.

**Comment by behrisch** [comment]
@m-kro Do we need another ticket for that (or reopen this one?)

**Comment by m-kro** [comment]
I would reopen the ticket as its title is still in line with the new findings.
